2025-05-02 12:25:24.378 | INFO     | datatrove.utils.logging:add_task_logger:58 - Launching pipeline for rank=0
2025-05-02 12:25:24.379 | INFO     | datatrove.utils.logging:log_pipeline:90 - 
--- 🛠️ PIPELINE 🛠
None: None
None: None
None: None
🔻 - FILTER: 🌍 Language ID
💽 - WRITER: 🐿 Jsonl
2025-05-02 12:25:27.775 | ERROR    | datatrove.executor.base:_run_for_rank:108 - 'DocumentWrapper' object is not subscriptable
Traceback (most recent call last):

  File "/Users/adrienloizeau/dev/scratchGPT/data_pipeline.py", line 59, in <module>
    executor.run()
    │        └ <function LocalPipelineExecutor.run at 0x11815db40>
    └ <datatrove.executor.local.LocalPipelineExecutor object at 0x118169a80>

  File "/Users/adrienloizeau/opt/anaconda3/envs/lerobot/lib/python3.10/site-packages/datatrove/executor/local.py", line 127, in run
    stats.append(self._launch_run_for_rank(rank, ranks_q))
    │     │      │    │                    │     └ <AutoProxy[Queue] object, typeid 'Queue' at 0x11816abf0>
    │     │      │    │                    └ 0
    │     │      │    └ <function LocalPipelineExecutor._launch_run_for_rank at 0x11815dab0>
    │     │      └ <datatrove.executor.local.LocalPipelineExecutor object at 0x118169a80>
    │     └ <method 'append' of 'list' objects>
    └ []
  File "/Users/adrienloizeau/opt/anaconda3/envs/lerobot/lib/python3.10/site-packages/datatrove/executor/local.py", line 76, in _launch_run_for_rank
    return self._run_for_rank(rank, local_rank)
           │    │             │     └ 0
           │    │             └ 0
           │    └ <function PipelineExecutor._run_for_rank at 0x11815d6c0>
           └ <datatrove.executor.local.LocalPipelineExecutor object at 0x118169a80>
> File "/Users/adrienloizeau/opt/anaconda3/envs/lerobot/lib/python3.10/site-packages/datatrove/executor/base.py", line 96, in _run_for_rank
    deque(pipelined_data, maxlen=0)
    │     └ <generator object DiskWriter.run at 0x118152ce0>
    └ <class 'collections.deque'>
  File "/Users/adrienloizeau/opt/anaconda3/envs/lerobot/lib/python3.10/site-packages/datatrove/pipeline/writers/disk_base.py", line 178, in run
    for document in data:
                    └ <generator object BaseFilter.run at 0x118153ed0>
  File "/Users/adrienloizeau/opt/anaconda3/envs/lerobot/lib/python3.10/site-packages/datatrove/pipeline/filters/base_filter.py", line 64, in run
    for batch in batched(data, self.batch_size):
                 │       │     │    └ 1
                 │       │     └ 🔻 - FILTER: 🌍 Language ID
                 │       └ <generator object ScoreFilter.run at 0x118152180>
                 └ <function batched at 0x118022d40>
  File "/Users/adrienloizeau/opt/anaconda3/envs/lerobot/lib/python3.10/site-packages/datatrove/utils/batching.py", line 20, in batched
    while batch := list(itertools.islice(it, n)):
                        │         │      │   └ 1
                        │         │      └ <generator object ScoreFilter.run at 0x118152180>
                        │         └ <class 'itertools.islice'>
                        └ <module 'itertools' (built-in)>

  File "/Users/adrienloizeau/dev/scratchGPT/data_pipeline.py", line 44, in run
    if int(document["score"]) > self.min_score:
           │                    │    └ 3
           │                    └ None: None
           └ <__main__.DocumentWrapper object at 0x11837cc70>

TypeError: 'DocumentWrapper' object is not subscriptable
